src/main/java/org/eeichinger/servicevirtualisation/jdbc/WireMockMappingJsonRecorder.java

```diff
@@ -0,0 +1,132 @@
+ */
+public class WireMockMappingJsonRecorder {
+
+    @Setter
+    private XmlTypeRegistry xmlTypeRegistry;
+
+    public WireMockMappingJsonRecorder() {
+        this(false);
+    }
+
+    public WireMockMappingJsonRecorder(boolean useExplicitTypes) {
+        if (useExplicitTypes) {
+            xmlTypeRegistry = new XmlTypeRegistry();
+        }
+    }
+
+    private int statementInTest = 0;
+
+    @SneakyThrows
+    protected String getSybaseXMLFromDBResult(ResultSet resultSet) {
+        StringBuilder parameterXml = new StringBuilder();
+        parameterXml.append("<resultset xmlns:xsi='http://www.w3.org/2001/XMLSchema-instance'><cols>");
+        ResultSetMetaData md = resultSet.getMetaData();
+        int numCols = md.getColumnCount();
+        List<String> colNames = IntStream.range(0, numCols)
+            .mapToObj(i -> {
+                try {
+                    return md.getColumnName(i + 1);
+                } catch (SQLException e) {
+                    e.printStackTrace();
+                    return "?";
+                }
+            }).collect(Collectors.toList());
+
+        colNames.forEach(s -> parameterXml
+            .append("<col>")
+            .append(s)
+            .append("</col>"));
+
+        parameterXml.append("</cols>");
+
+        while (resultSet.next()) {
+            parameterXml.append("<row>");
+            for (String cn : colNames) {
+                Object rowValue = resultSet.getObject(cn);
+                if (rowValue == null) {
+                    parameterXml.append("<val xsi:nil='true'></val>");
+                } else {
+                    if (xmlTypeRegistry != null) {
+                        String mappedType = xmlTypeRegistry.getXmlType(rowValue.getClass());
+                        if (mappedType != null) {
+                            parameterXml.append("<val xsi:type='").append(mappedType).append("'>").append(rowValue).append("</val>");
+                        } else if (rowValue instanceof String) {
+                            parameterXml.append("<val>").append(rowValue).append("</val>");
+                        } else {
+                            throw new RuntimeException("Didn't know how to write out " + rowValue + " of class " + rowValue.getClass());
+                        }
+                    } else { //always coerce to string
+                        parameterXml.append("<val>").append(rowValue).append("</val>");
+                    }
+                }
+            }
+            parameterXml.append("</row>");
+        }
+
+        parameterXml.append("</resultset>");
+
+        return parameterXml.toString();
+    }
+
+    protected String getRecordFilePath(int statementInTest) {
+        return "src/test/resources/newMappings/" + statementInTest + ".json";
+    }
+
+    @SneakyThrows
+    public void writeOutMapping(JdbcServiceVirtualizationFactory.PreparedStatementInformation preparedStatementInformation, ResultSet resultSet) {
```

Comment:
Some terrible and lazy templating, but I didn't want to bring in a dependency or anything. I expect others using this functionality will want to roll their own from scratch anyway.